src/Controls/src/Core/Handlers/Shell/ShellSectionHandler.Windows.cs

```diff
@@ -45,7 +45,9 @@ public override void SetVirtualView(Maui.IElement view)
 			if (_shellSection != null)
 			{
 				((IShellSectionController)_shellSection).NavigationRequested -= OnNavigationRequested;
-				((IShellController)_shellSection.FindParentOfType<Shell>()!).RemoveAppearanceObserver(this);
+
+				var shell = _shellSection.FindParentOfType<Shell>() as IShellController;
+				shell?.RemoveAppearanceObserver(this);
```

Comment:
I think we need to store shell locally when the observer is set on line 75 that way we can successfully remove (this) from the list of AppearanceObservers. Otherwise, this Handler will probably remain inside the list of appearance observers inside shell.

https://github.com/dotnet/maui/blob/f6cf7c78b48fdca0a0249be4f631f1f6ba8e2fd0/src/Controls/src/Core/Shell/Shell.cs#L313

Or trigger this tear down code via
https://github.com/dotnet/maui/blob/f6cf7c78b48fdca0a0249be4f631f1f6ba8e2fd0/src/Core/src/Platform/ActionDisposable.cs#L8